### ISSUE TITLE
wfevent crash on win64

### ISSUE THREAD

**Issue by zack-vii** [MDSplus/mdsplus#77]
A client running on win 64bits will crash on wfevent.fun once the event occurs.
Thereby the source of the event (remote/local, 32/64bit, win/unix) does not seem to play a role.
jScope works fine though. I suppose it has a direct link to the MDSplus event core.
Since MATLAB is also based on java, it might be caused by the TDI interface.

**Comment by tfredian** [comment]
I can't seem to replicate this problem. I did a wfevent() in a tdi 
window and a setevent in a cmd.exe window and it worked with or without 
data in the event. Do you have a test that exhibits this problem? I may 
not be running the same version as you. Let me know which version you 
have installed an I can test that too.

-tom

**Comment by zack-vii** [comment]
Well, I just installed MDSplus-alpha-7.0.147 on a clean Windows XP 32bit VM.
If I open two tditest sessions and execute in first one

``` tdi
wfevent('e')
```

it crashes ("0x77c17631", "0x003ed000", "written"), once I execute

``` tdi
setevent('e')
```

in the second one.

FYI: the application wfevent.exe works without problems

**Comment by zack-vii** [comment]
In python-2.7.10 amd64 executing

``` python
MDSplus.Event.wfevent('e')
```

throws (when the event occurs)

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~\python-2.7.10.amd64\Lib\site-packages\MDSplus\event.py", line 93, in wfevent

  File "~\python-2.7.10.amd64\Lib\site-packages\MDSplus\mdsarray.py", line 195, in deserialize
    """
  File "~\python-2.7.10.amd64\Lib\site-packages\MDSplus\mdsdata.py", line 806, in deserialize
    """Return MDSplus data class from value.
  File "~\python-2.7.10.amd64\Lib\site-packages\MDSplus\_mdsshr.py", line 116, in MdsSerializeDscIn
    if (status & 1) == 1:
WindowsError: exception: access violation reading 0x000000008FD6FF20
```

**Comment by zack-vii** [comment]
after applying the version stuff I testes the events again.. it does not crash anymore. However it is still buggy:
1. after an event has occurred the wait thread of wfevent is not closed properly and its cpu time occupies a whole core. this accumulates with subsequent calls of wfevent

``` python
MDSplus.Event.wfevent('e')
MDSplus.Event.setevent('e')
```
1. the return value of wfevent, bytes in MdsSerializeDscIn, seems to have some kind of header. However the header is missing if setevent is executed with only one argument. MdsSerializeDscIn will raise an error if not handled e.g. like in  9f71356 #94 
2. If setevent did send an array, MdsSerializeDscIn will cut out the correct array - MATLAB, however, does not. Also MATLAB has the thread problem (1.).

```
1 , 0 , 6 , 4 , 16 , 0 , 0 , 0 ,0 , 0 , 0 , 1  ,  3 , 0 , 0 , 0  ,  1 , 2 , 3
^------------------header-------------------^     ^-length-4B-^    ^--array--^
```

**Comment by tfredian** [comment]
Does this problem occur just on Windows? I haven't been able to reproduce it on linux. Unless you are using python to set the event you should be using wfeventRaw. Originally events only passed one or more bytes of data with a max of 16 bytes. This limit was expanded with the addition of udp events but all applications with the exception of the python module only send data as a series of bytes. The setevent and wfevent functions in python were enhanced to send serialized data so you could pass more complex data items and inspect them on the receiving end to see what you received. If you are receiving event data being sent from other languages you would use wfeventRaw which just gets the data as a series of bytes if any are included when setting the event.

Please provide the platform (i.e. windows xp 32-bit) and an example which produces this problem so we can replicate it to figure out what is happening.

**Comment by zack-vii** [comment]
got fixed by e990b5a
